aiida_quantumespresso/workflows/protocols/utils.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""Utilities to manipulate the workflow input protocols."""
+import functools
+import os
+import pathlib
+import yaml
+
+
+def recursive_merge(left, right):
+    """Recursively merge two dictionaries into a single dictionary.
+
+    :param left: first dictionary
+    :param right: second dictionary
+    :return: the recursively merged dictionary
+    """
+    import collections
+
+    for key, value in left.items():
+        if key in right:
+            if isinstance(value, collections.abc.Mapping) and isinstance(right[key], collections.abc.Mapping):
+                right[key] = recursive_merge(value, right[key])
+
+    merged = left.copy()
+    merged.update(right)
+
+    return merged
+
+
+def load_protocol_file(cls):
+    """Load the protocol file for the given workflows class.
+
+    :param cls: the workflow class.
+    :return: the contents of the protocol file.
+    """
+    from aiida.plugins.entry_point import get_entry_point_from_class
+
+    _, entry_point = get_entry_point_from_class(cls.__module__, cls.__name__)
+    entry_point_name = entry_point.name
+    parts = entry_point_name.split('.')
+    parts.pop(0)
+    filename = f'{parts.pop()}.yaml'
+    basepath = functools.reduce(os.path.join, parts)
```

Comment:
Sure, but this is fancier ;) plus it would be easier to refactor once `parts` became `pathlib.Path` parts. Then you just replace `os.path.join` with the `/` operator.